pandas/tests/plotting/test_datetimelike.py

```diff
@@ -253,7 +256,7 @@ def test_line_plot_datetime_frame(self, freq):
             index=idx,
             columns=["A", "B", "C"],
         )
-        freq = freq_to_period_freqstr(1, df.index.freq.rule_code)
+        freq = re.split(r"\d", PeriodDtype(df.index.freq)._freqstr)[-1]
```

Comment:
why does this one require regex splitting, but others don't?

Reply:
Hmm, seems like we  don't need this regex splitting here and can replace `freq = re.split(r"\d", PeriodDtype(df.index.freq)._freqstr)[-1]` with `freq = PeriodDtype(df.index.freq)._freqstr`

I will make this change.

There is another test `test_line_plot_period_mlt_frame` in which we use the same regex splitting. I tried to do the similar replacement (with `freq = PeriodDtype(df.index.freq)._freqstr[1:]`), but the test failed for `"1s"`.

Reply:
I think in `test_line_plot_period_mlt_frame` we can change `freq`  from
```
    freq = re.split(r"\d", PeriodDtype(df.index.freq)._freqstr)[-1]
    freq = df.index.asfreq(freq).freq
```
to 
```
freq = df.index.freq.rule_code
```
and then get rid of regex splitting. I modified the test.

Reply:
if we use the same replacement in `test_line_plot_datetime_frame`, e.g. replace 
```
freq = PeriodDtype(df.index.freq)._freqstr
freq = df.index.to_period(freq).freq
```
with
```
freq = df.index.freq.rule_code
```
the test passes. Should we modify this test as well?